File: SVR/support_vector_regression.py
# Polynomial regression

# Import libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Import dataset
dataset = pd.read_csv('Position_Salaries.csv')

X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

# Feature scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
sc_y = StandardScaler()
X = sc_X.fit_transform(X)
y = sc_y.fit_transform(y)

# Support Vector Regression model
from sklearn.svm import SVR
regressor = SVR(kernel = 'rbf')
regressor.fit(X, y)

# Visualize SVR regression
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, y, color = "red")
plt.plot(X_grid, regressor.predict(X_grid), color = "blue")
plt.title("SVR model fit")
plt.xlabel("Position index")
plt.ylabel("Salary")
plt.show()

# Prediction with SVR model
y_pred = sc_y.inverse_transform(regressor.predict(sc_X.transform(np.array([6.5]))))

File: SVR/support_vector_regression.R
# SVR

# Importing the dataset
dataset <- read.csv('Position_Salaries.csv')
dataset <- dataset[2:3]

# Support Vector Regression
library(e1071)
regressor = svm(Salary ~ Level, data = dataset, type = "eps-regression")

# Visualising the polynomial model results (for higher resolution and smoother curve)
library(ggplot2)
x_grid <- seq(min(dataset$Level), max(dataset$Level), 0.1)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             colour = 'red') +
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))),
            colour = 'blue') +
  ggtitle('SVR model') +
  xlab('Level') +
  ylab('Salary')

# SVR prediction
y_pred <- predict(regressor, data.frame(Level = 6.5))
